### DISCUSSION TITLE
LAVA vs. labgrid

### DISCUSSION THREAD

**Discussion by davidmaze1990** [labgrid-project/labgrid#1139] [Q&A] [answered]
Hi, there,

When search automation framework, I saw lava and then I got labgrid. I see you guys also have lava used: https://lava.pengutronix.de/, so I wonder how you internal use them? Different projects or different scenario using different framework?

I'd like to listen to you about best practice to choose between them.

Thanks,
David

**Comment by Emantor** [comment]
Hi,

> Hi, there,
> 
> When search automation framework, I saw lava and then I got labgrid. I see you guys also have lava used: https://lava.pengutronix.de/, so I wonder how you internal use them? Different projects or different scenario using different framework?

Yes, exactly. We are running a lava lab for kernelci, but all our internal project testsuites are based on labgrid. Lava is great if the same programs need to be run on many boards in linux userspace, however last time I checked it was hard to integrate more complex configurations such as FPGA boards into the test flow. Thats one of the reasons why we developed labgrid instead of using lava for every testsuite.

> I'd like to listen to you about best practice to choose between them.

This likely depends more on your individual usecase. Do you have some examples?

**Comment by davidmaze1990** [comment]
@Emantor , we mainly run linux LTP & android xts, so for this scenario, looks both lava & labgrid can support it, what would be the best selection?

**Reply by jluebbe** [reply, marked as answer]
For those testsuites, there are probably more existing integrations with LAVA than with Labgrid. With labgrid, you can also easily use the same boards for interactive development.
